targets/targets.json

```diff
@@ -3658,6 +3658,9 @@
     "NRF52840_DK": {
         "supported_form_factors": ["ARDUINO"],
         "inherits": ["MCU_NRF52840"],
+        "macros_add": ["MBEDTLS_CONFIG_HW_SUPPORT"],
+        "device_has_add": ["CRYPTOCELL"],
```

Comment:
This is HAL implementation for cryptocell, that we provide. And then what is the extra label for ? To specify the version?

Reply:
I added the `DEVICE_CRYPTOCELL` label for the sake of using the cryptocell trng.
For example, in the platform's [`trng_api.c`](https://github.com/ARMmbed/mbed-os/pull/6794/files#diff-e9952cc585c4b9194267c89d0b9e7390R40), it is mutual to all of NRF5x boards, but on the NRF52840_DK, there is also cryptocell TRNG.
I am thinking that perhaps in the future, to use this label also for the MPS2 porting, done in #6169 but this is only if the API will be common for both targets, which is currently not.

The label for `DEVICE_CRYPTOCELL` is more intended to the partner's code, for them to specify Cryptocell specific function( in case it is common for several boards)

Reply:
If your question is about the `TARGET_CRYPTOCELL310`, this is specifically the CC310 product. Future Cryptocell products will have a different target

Reply:
`DEVICE_` is used for HAL, device capabilities, like `trng`. Shouldn't this just be part of it? If cryptocell is enabled, it is used instead ? Does this new cryptocell provide own API or it's just what trng should use? Or does it need better name `CRYPTOCELL_TRNG` ?

I am confused what cryptocell is , why is it being defined as `DEVICE_CRYPTOCELL` and needs extra label of its own to specify a version

Reply:
Cryptocell is Arm's HW accelerator.
perhaps `DEVICE_CRYPTOCELL` is not the right MACRO, but I used `device_has`, same way the MICROVISOR was used.

The extra label, is the actual target version, that partner should add\enable.
The `DEVICE_CRYPTOCELL` is in compilation time, to determine in the partner's common code, for some functionality that should be compiled in \ out of the partner's specific code contains Cryptocell HW. I wouldn't limit it only for TRNG, but this is probably the most common case.
The Cryptocell trng implements same API for trng as any other trng driver, so I needed some special compilation flag to determine the specific board has cryptocell, to avoid multiple definition of same function.

Perhaps the Readme I just added could explain better.

Reply:
Is cryptocell part of the SoC or is it something separate? That you connect to the board/design?

Reply:
Cryptocell is an ARM security module that is part of the SoC.
The device brief mentions it several times: https://infocenter.nordicsemi.com/pdf/nRF52840_PB_v2.0.pdf

@0xc0170 I think the problem with `CRYPTOCELL_TRNG` is that CRYPTOCELL appears to be more generic than simply applying to the TRNG.

Reply:
Cryptocell is Arm's HW crypto accelerator. It is more than just TRNG, this is why I defined it as `DEVICE_CRYPTOCELL`, to add flexibility for the partner to decide whether to use Cryptocell or their code, if there are duplicates